src/libraries/Microsoft.PowerFx.Connectors/Tabular/Services/CdpTable.cs

```diff
@@ -61,10 +60,19 @@ internal CdpTable(string dataset, string table, DatasetMetadata datasetMetadata,
         }
 
         //// TABLE METADATA SERVICE
-        // GET: /$metadata.json/datasets/{datasetName}/tables/{tableName}?api-version=2015-09-01
+        // GET: /$metadata.json/datasets/{datasetName}/tables/{tableName}?api-version=2015-09-01        
         public virtual async Task InitAsync(HttpClient httpClient, string uriPrefix, CancellationToken cancellationToken, ConnectorLogger logger = null)
         {
             cancellationToken.ThrowIfCancellationRequested();
+#pragma warning disable CS0618 // Type or member is obsolete
+            await InitAsync(httpClient, uriPrefix, TableName, cancellationToken, logger).ConfigureAwait(false);
+#pragma warning restore CS0618 // Type or member is obsolete
+        }
+
+        [Obsolete("This API should not be called directly - used by CDP Validator only", false)]
```

Comment:
>  [Obsolete("This API should not be called directly - used by CDP Validator only", false)] [](http://example.com/codeflow?start=6&length=90)

why only via validator? 
It's reasonable to go from TableName --> CdpTable

Reply:
We have CdpDataSource.GetTableAsync but it will call CDP GetTables API and identiify if a table name is invalid

Reply:
Why doesn't CdpDataSource.GetTableAsync just call this API?  (why does it call all tables, and then filter to the one requested ... instead of just calling the API for the one requested)